### PR TITLE
id3 improvements

### PR DESCRIPTION
**Pull request by rbouqueau**
This PR should improve the id3 support and revert some breaking changes that happened a couple of months ago. id3 stuff moved in its own file.

The only thing I dislike is the new ```OFFS(inband_event)``` option in the dasher to signal ID3 presence in the DASH manifest. The issue is that some markers require some custom input in the manifest e.g. Nielsen's require both:
```
  <InbandEventStream schemeIdUri="https://aomedia.org/emsg/ID3" value="www.nielsen.com:id3:v1"/>
  <InbandEventStream schemeIdUri="https://aomedia.org/emsg/ID3" value="https://aomedia.org/emsg/ID3"/>
```

Maybe some improvements when processing #2828 can add further clarity. How can we add data to the manifest based on packet data?

### CODE REVIEW

**Comment by aureliendavid** [comment]
Hi, 

thanks for the pr

can you merge / rebase master into it, so that you get the fix in 255b03e3805fba53011e28be0c776122142fbfbf to make the test results easier to interpret?

ty

**Comment by rbouqueau** [comment]
Thanks @aureliendavid . Just did it.

File: src/filters/dasher.c [hunk 1]
```diff
@@ -2865,6 +2866,26 @@ static void dasher_add_descriptors(GF_List **p_dst_list, const GF_PropertyValue
 	}
 }
 
+static void dasher_add_inband_event(GF_DashStream *ds)
+{
+	GF_MPD_Inband_Event *nielsen_event;
+	GF_MPD_Inband_Event *custom_event;
+	if(ds->stream_type == GF_STREAM_AUDIO) {
+		GF_SAFEALLOC(custom_event, GF_MPD_Inband_Event);
+		custom_event->scheme_id_uri = gf_strdup("https://aomedia.org/emsg/ID3");
```
Comment (jadarve) [inline src/filters/dasher.c:2875]: Is there an issue tracking the generalization of this flag?

The `scheme_id_uri` and `value` could be extracted from the `GF_ID3_TAG` properties attached to the audio packets.

Comment (rbouqueau) [reply src/filters/dasher.c:2875]: That's a good idea. Do you feel like doing it before merging or should we open a separate issue?

Comment (jadarve) [reply src/filters/dasher.c:2875]: Better to open a separate issue. We can try to solve it at a later time. @sebastianBD95 @gomezpirry

**Comment by rbouqueau** [comment]
@jeanlf @aureliendavid Good to go!

**Comment by jeanlf** [comment]
thanks!

**Pull request merged**